src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -161,25 +188,24 @@ private float addRow(
 
     // Create new empty stats just for the current row.
     Map<String, RowBufferStats> forkedStatsMap = new HashMap<>();
+    statsMap.forEach((columnName, stats) -> forkedStatsMap.put(columnName, stats.forkEmpty()));
```

Comment:
we don't want to do unnecessary iterations, please keep the old inlined `forkedStatsMap.put` in place.

Reply:
i see IcebergParquetValueParser needs the whole map, not sure yet why that is, still reading through the changes.

Reply:
resolving this.